File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/parser/scala3/Scala3ParserTest.java
package org.jetbrains.plugins.scala.lang.parser.scala3;

import com.intellij.lang.Language;
import junit.framework.Test;
import junit.framework.TestCase;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.plugins.scala.Scala3Language;
import org.jetbrains.plugins.scala.lang.parser.ScalaFileSetParserTestCase;

public class Scala3ParserTest extends TestCase {
    @NotNull
    public static Test suite() {
        // TODO: test: change from data3 to data and check all failures, are they expected?
        //  if yes, add a dedicated test for Scala3 with a different expected result
        return new ScalaFileSetParserTestCase("/parser/data3") {
            @NotNull
            @Override
            protected Language getLanguage() {
                return Scala3Language.INSTANCE;
            }
        };
    }
}

File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/parser/scala3/Scala3ParserTest.scala
package org.jetbrains.plugins.scala.lang.parser.scala3

import com.intellij.lang.Language
import org.jetbrains.plugins.scala.Scala3Language
import org.jetbrains.plugins.scala.base.{DefaultFileSetTestTransform, NoSdkFileSetTestBase}

import java.nio.file.Path

/**
 * @see [[org.jetbrains.plugins.scala.lang.parser.ScalaParserTest]]
 */
class Scala3ParserTest extends NoSdkFileSetTestBase with DefaultFileSetTestTransform {
  override protected def relativeTestDataPath: Path = Path.of("parser", "data3")

  override def language: Language = Scala3Language.INSTANCE
}
